plugins/by-name/typescript-tools/default.nix

```diff
@@ -160,43 +160,69 @@ lib.nixvim.neovim-plugin.mkNeovimPlugin {
 
   # NOTE: call setup in lsp config
   callSetup = false;
-  extraConfig = cfg: {
-    plugins.lsp.postConfig = ''
-      require('typescript-tools').setup(${lib.nixvim.toLuaObject cfg.settings})
-    '';
-
-    # TODO:: introduced 10-22-2024: remove after 24.11
-    # Nested settings can't have normal mkRenamedOptionModule functionality so we can only
-    # alert the user that they are using the old values
-    warnings =
-      let
-        definedOpts = lib.filter (opt: lib.hasAttrByPath (lib.toList opt) cfg.settings) [
-          "separateDiagnosticServer"
-          "publishDiagnosticOn"
-          "exposeAsCodeAction"
-          "tsserverPath"
-          "tsserverPlugins"
-          "tsserverMaxMemory"
-          "tsserverFormatOptions"
-          "tsserverFilePreferences"
-          "tsserverLocale"
-          "completeFunctionCalls"
-          "includeCompletionsWithInsertText"
-          "codeLens"
-          "disableMemberCodeLens"
-          [
-            "jsxCloseTag"
-            "enable"
-          ]
-          [
-            "jsxCloseTag"
-            "filetypes"
-          ]
-        ];
-      in
-      lib.optional (definedOpts != [ ]) ''
-        Nixvim(plugins.typescript-tools): The following settings have moved under `plugins.typescript-tools.settings.settings` with snake_case:
-        ${lib.concatMapStringsSep "\n" (opt: "  - ${lib.showOption (lib.toList opt)}") definedOpts}
+  extraConfig =
+    let
+      oldSettings = [
+        "separateDiagnosticServer"
+        "publishDiagnosticOn"
+        "exposeAsCodeAction"
+        "tsserverPath"
+        "tsserverPlugins"
+        "tsserverMaxMemory"
+        "tsserverFormatOptions"
+        "tsserverFilePreferences"
+        "tsserverLocale"
+        "completeFunctionCalls"
+        "includeCompletionsWithInsertText"
+        "codeLens"
+        "disableMemberCodeLens"
+        [
+          "jsxCloseTag"
+          "enable"
+        ]
+        [
+          "jsxCloseTag"
+          "filetypes"
+        ]
+      ];
+
+    in
+    cfg: {
+      plugins.lsp.postConfig = ''
+        require('typescript-tools').setup(${lib.nixvim.toLuaObject cfg.settings})
       '';
-  };
+
+      # TODO:: introduced 10-22-2024: remove after 24.11
+      # Nested settings can't have normal mkRenamedOptionModule functionality so we can only
+      # alert the user that they are using the old values
+      plugins.typescript-tools.settings.settings =
+        let
+          definedOpts = builtins.trace "Defined options in cfg.settings" (
+            lib.filter (opt: lib.hasAttrByPath (lib.toList opt) cfg.settings) oldSettings
+          );
+
+          migratedSettings = builtins.trace "Migrated settings" (
+            lib.fold (
+              acc: oldName:
+              let
+                newPath = lib.nixvim.toSnakeCase oldName;
```

Comment:
I don't think this handles list-paths correctly? `toSnakeCase` is type String -> String

https://github.com/nix-community/nixvim/blob/029eafd70d6e28919a9ec01a94a46b51c4ccff40/lib/utils.nix#L64-L77

Reply:
I think you want:
```nix
let
  oldPath = toList oldName;
  newPath = map toSnakeCase oldPath;
in
```

Reply:
I thought I had that in here at one point... must have reverted it trying to figure out the infinite recursion.